daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -718,3 +721,90 @@ class UnparseResult(dp: DataProcessor, ustate: UState)
     encodingInfo.knownEncodingName
   }
 }
+
+class DaffodilXMLReader(dp: DataProcessor) extends XMLReader {
+  private var _contentHandler: ContentHandler = null
+  private var _errorHandler: ErrorHandler = null
+  private var _dtdHandler: DTDHandler = null
+  private var _entityResolver: EntityResolver = null
+  val SAXNamespaceFeature = "http://xml.org/sax/features/namespaces"
+  val SAXNamespacePrefixFeature = "http://xml.org/sax/features/namespace-prefixes"
+  private var _featureMap = mutable.HashMap[String, Boolean](SAXNamespaceFeature -> false,
+    SAXNamespacePrefixFeature -> false)
+
+  override def getFeature(name: String): Boolean = {
+    name match {
+      case SAXNamespaceFeature | SAXNamespacePrefixFeature => _featureMap(name)
+      case _ => {
+        throw new SAXNotRecognizedException("Only namespace and namespace features are supported for SAX parsing")
+        false
+      }
+    }
+  }
+
+  override def setFeature(name: String, value: Boolean): Unit = {
+    name match {
+      case SAXNamespaceFeature | SAXNamespacePrefixFeature => _featureMap(name) = value
+      case _ => {
+        throw new SAXNotRecognizedException("Only namespace and namespace features are supported for SAX parsing")
+      }
+    }
+  }
+
+  override def getProperty(name: String): AnyRef = {
+    throw new SAXNotRecognizedException("SAX Properties unsupported")
+  }
+
+  override def setProperty(name: String, value: Any): Unit = {
+    throw new SAXNotRecognizedException("SAX Properties unsupported")
+  }
+
+  override def setEntityResolver(resolver: EntityResolver): Unit = {
+    _entityResolver = resolver
+  }
+
+  override def getEntityResolver: EntityResolver = _entityResolver
+
+  override def setDTDHandler(handler: DTDHandler): Unit = {
+    _dtdHandler = handler
+  }
+
+  override def getDTDHandler: DTDHandler = _dtdHandler
+
+  override def setContentHandler(handler: ContentHandler): Unit = {
+    _contentHandler = handler;
+  }
+
+  override def getContentHandler: ContentHandler = _contentHandler
+
+  override def setErrorHandler(handler: ErrorHandler): Unit = {
+    _errorHandler = handler;
+  }
+
+  override def getErrorHandler: ErrorHandler = _errorHandler
+
+  override def parse(input: InputSource): Unit = {
+    val is = input.getByteStream
+    if(is != null) {
+      parse(is)
+    } else {
+      throw new IOException("Inputsource must be backed by Inputstream")
+    }
+  }
+
+  override def parse(systemId: String): Unit = {
+    throw new IOException("SAX parsing of systemId is unsupported")
+  }
+
+  def parse(stream: InputStream): DFDL.ParseResult = {
+    val isdis = InputSourceDataInputStream(stream)
+    val sio = new SAXInfosetOutputter(_contentHandler)
+    dp.parse(isdis, sio)
+  }
+
+  def parse(arr: Array[Byte]): DFDL.ParseResult = {
+    val isdis = InputSourceDataInputStream(arr)
+    val sio = new SAXInfosetOutputter(_contentHandler)
+    dp.parse(isdis, sio)
+  }
+}
```

Comment:
Instead of these other parse methods returning a ParseResult, they should probably return a Unit just the the standard XMLReader.parse method. With SAX, they way errors are determined is by setting an error handler. So probably what we need to do get the ParseResult and call appropriate functions in the errorHandler to send diagnostics via warn/error/fatalError. If they user doesn't set an errorHandler then errors a just silently ignored. We might also want to throw a SAXException or IOException as a more obvious sign that the parse failed. I'm not sure how SAX parsing is expected to handle errors. Maybe somethign to look into more.